deep_qa/training/trainer.py

```diff
@@ -319,7 +319,7 @@ def train(self):
         # so you may have left it above zero on accident.
         if self.validation_arrays is not None:
             kwargs['validation_data'] = self.validation_arrays
-        elif self.validation_split > 0.0:
+        elif self.validation_split > 0.0 and isinstance(self.training_arrays, tuple):
```

Comment:
Hmm, yeah, you're right.  Trying to remember why I didn't do that in the first place...

Oh, it was because `self.use_data_generator` is only defined in `TextTrainer`, not here.  I thought it wasn't a great idea to have a parameter to `Trainer` that `Trainer` can't actually do anything about - the `Trainer` class doesn't know anything about how data is generated, and you could imagine a subclass that doesn't implement data generators - how should `Trainer` handle the `use_data_generator` parameter in that case?

Anyway, I think the right way to fix this is to add a field here, not a parameter, like `uses_data_generator`, and check that instead of having type checks for `tuple`.  You're right that this would be _way_ more readable.  Subclasses will have to set that field in their constructor, after the call to `super().__init__`.  I'll make this change before merging.